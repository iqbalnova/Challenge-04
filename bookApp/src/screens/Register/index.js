import { View, Text, Image, StyleSheet, Alert, RefreshControl, ActivityIndicator } from 'react-native'
import React, { useState } from 'react'
import { ScrollView, TouchableOpacity } from 'react-native-gesture-handler'
import { backdropRegis } from '../../assets/img'
import { ms } from 'react-native-size-matters'
import { Fumi } from 'react-native-textinput-effects';
import Ionicons from 'react-native-vector-icons/Ionicons'
import Poppins from '../../components/Poppins'
import Feather from 'react-native-vector-icons/Feather'
import { useDispatch, useSelector } from 'react-redux'
import { setLoading } from '../../redux/globalAction'
import axios from 'axios'
import { BASE_URL } from '../../helpers/API'

export default function Register({navigation}) {

  const [fullname, setFullName] = useState(''); // Sampe Sahur
  const [password, setPassword] = useState(''); // sampesahur23
  const [email, setEmail] = useState(''); // muhammadiqbal2354@gmail.com

  const dispatch = useDispatch();
  const {loading} = useSelector(state => state.global);

  const regexEmail = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
  const regexPassword = /^[a-z0-9]\w{8,14}$/;

  const postRegister = async () => {
    if(fullname.length === 0){
      Alert.alert('Warning', 'FullName harus diisi!')
    }
    else if(email.length === 0){
      Alert.alert('Warning', 'Email harus diisi!');
    } 
    else if(password.length === 0){
      Alert.alert('Warning', 'Password harus diisi!')
    }
      else{
        if (!email.match(regexEmail)){
          alert('Email tidak valid')
        }
        else if (!password.match(regexPassword)){
          alert('Password tidak valid. Minimal 8 karakter dan 1 angka')
        } 
          else {
            
              try {
                dispatch(setLoading(true));
                const body = {
                  email: email,
                  password: password,
                  name: fullname
                };
                const res = await axios.post(`${BASE_URL}/auth/register`,body, {
                  validateStatus: status => status < 501});
                console.log(res);
              
                if(res.status <= 201){
                  navigation.navigate("Success")
                } else {
                  Alert.alert("Error", "Tidak bisa register")
                }
              } catch (error) {
                console.log(error);
              } finally{
                dispatch(setLoading(false));
              }
          }
        }
  } 
  
  const [refreshing, setRefreshing] = useState(false);
  const onRefresh = () =>{
    setRefreshing(true);
    setTimeout(() => {
      setRefreshing(false)
      
    }, 2000);
  }
  
  return (
    <ScrollView 
    refreshControl={
      <RefreshControl refreshing={refreshing} onRefresh={onRefresh} />
    }
    style={styles.container}>
      <View style={styles.imgContainer}>
        <Image source={backdropRegis} style={styles.img}/>
      </View>
      <View style={{ marginHorizontal: ms(20), marginTop:ms(30) }}>
        <Fumi
            label={'Full Name'}
            iconClass={Feather}
            iconName={'user'}
            iconColor={'#f95a25'}
            iconSize={ms(20)}
            iconWidth={ms(40)}
            inputPadding={ms(16)}
            onChangeText={(value) => setFullName(value)}
          />
      </View>
      <View style={{ marginHorizontal: ms(20), marginTop:ms(10) }}>
        <Fumi
          label={'E-mail'}
          iconClass={Ionicons}
          iconName={'mail-outline'}
          iconColor={'#f95a25'}
          iconSize={ms(20)}
          iconWidth={ms(40)}
          inputPadding={ms(16)}
          onChangeText={(value) => setEmail(value)}
        />
      </View>
      <View style={{ marginHorizontal: ms(20), marginTop:ms(10) }}>
        <Fumi
            label={'Password'}
            iconClass={Ionicons}
            iconName={'lock-closed-outline'}
            iconColor={'#f95a25'}
            iconSize={ms(20)}
            iconWidth={ms(40)}
            inputPadding={ms(16)}
            secureTextEntry={true}
            onChangeText={(value) => setPassword(value)}
          />
      </View>
      {loading ? <ActivityIndicator /> :
      <TouchableOpacity style={styles.button} onPress={() => postRegister()}>
        <Text style={{textAlign:"center", color: '#fff', fontSize: ms(16)}}>
          Register
        </Text>
      </TouchableOpacity>}
      <View style={{ justifyContent:'center', alignItems: 'center', marginTop: ms(10), marginBottom: ms(5) }}>
        <Poppins size={ms(14)}>Already have an account?</Poppins>
      </View>
      
      <TouchableOpacity onPress={() => navigation.navigate('Login')}>
        <Text style={{textAlign: 'center', color: 'blue', fontSize:ms(16), fontWeight:'bold'}}>Login</Text>
      </TouchableOpacity>
    </ScrollView>
  )
}

const styles = StyleSheet.create({
  container:{
    flex: 1,
    backgroundColor: '#a7cbad'
  },
  imgContainer:{
    justifyContent: 'center',
    alignItems: 'center', 
    marginHorizontal: ms(20),
    marginTop: ms(50)
  },
  img : {
    width: ms(330),
    height: ms(200),
    resizeMode:'cover',
  },
  button:{
    backgroundColor: '#659cff',
    marginTop: ms(10),
    marginHorizontal: ms(20),
    paddingVertical: ms(15),
    borderRadius: ms(5)
  }
})
