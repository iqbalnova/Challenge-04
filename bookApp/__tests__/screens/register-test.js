import React from 'react';
import {create} from 'react-test-renderer';
import Register from '../../src/screens/Register';
import ContainerTesting from '../../src/helpers/ReduxTesting';
// const tree = create(<Home />);

// test('snapshot', () => {
//   expect(tree).toMatchSnapshot();
// });
jest.runAllTimers();
describe('Group Name', () => {
  describe('Register', () => {
    it('renders correctly', () => {
      const tree = create(ContainerTesting(<Register />));
      expect(tree).toMatchSnapshot();
    });
  });
});
