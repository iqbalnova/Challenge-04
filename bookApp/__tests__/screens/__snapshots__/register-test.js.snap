// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group Name Register renders correctly 1`] = `
<RCTScrollView
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
  style={
    Object {
      "backgroundColor": "#a7cbad",
      "flex": 1,
    }
  }
>
  <RCTRefreshControl />
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginHorizontal": 31.428571428571427,
          "marginTop": 78.57142857142857,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/assets/img/backdropRegis.jpg",
          }
        }
        style={
          Object {
            "height": 314.2857142857143,
            "resizeMode": "cover",
            "width": 518.5714285714286,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "marginHorizontal": 31.428571428571427,
          "marginTop": 47.14285714285714,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "overflow": "hidden",
              "paddingTop": 16,
            },
            undefined,
            Object {
              "height": 73.14285714285714,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#f95a25",
                "fontSize": 31.428571428571427,
              },
              Object {
                "bottom": 24,
                "color": "rgba(163, 163, 163, 1)",
                "left": 25.142857142857142,
                "position": "absolute",
              },
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#f0f0f0",
                "position": "absolute",
                "top": 8,
                "width": 1,
              },
              Object {
                "height": 48,
                "left": 70.85714285714286,
              },
            ]
          }
        />
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 48,
              "left": 88,
              "position": "absolute",
              "top": 24,
            }
          }
        >
          <Text
            collapsable={false}
            style={
              Object {
                "color": "rgba(105, 105, 105, 1)",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Full Name
          </Text>
        </View>
        <TextInput
          animationDuration={300}
          height={48}
          iconClass={[Function]}
          iconColor="#f95a25"
          iconName="user"
          iconSize={31.428571428571427}
          iconWidth={62.857142857142854}
          inputPadding={25.142857142857142}
          label="Full Name"
          onBlur={[Function]}
          onChange={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          passiveIconColor="#a3a3a3"
          style={
            Array [
              Object {
                "color": "black",
                "flex": 1,
                "fontSize": 18,
                "padding": 7,
                "paddingLeft": 0,
              },
              Object {
                "color": "#f95a25",
                "marginLeft": 88,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginHorizontal": 31.428571428571427,
          "marginTop": 15.714285714285714,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "overflow": "hidden",
              "paddingTop": 16,
            },
            undefined,
            Object {
              "height": 73.14285714285714,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#f95a25",
                "fontSize": 31.428571428571427,
              },
              Object {
                "bottom": 24,
                "color": "rgba(163, 163, 163, 1)",
                "left": 25.142857142857142,
                "position": "absolute",
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#f0f0f0",
                "position": "absolute",
                "top": 8,
                "width": 1,
              },
              Object {
                "height": 48,
                "left": 70.85714285714286,
              },
            ]
          }
        />
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 48,
              "left": 88,
              "position": "absolute",
              "top": 24,
            }
          }
        >
          <Text
            collapsable={false}
            style={
              Object {
                "color": "rgba(105, 105, 105, 1)",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            E-mail
          </Text>
        </View>
        <TextInput
          animationDuration={300}
          height={48}
          iconClass={[Function]}
          iconColor="#f95a25"
          iconName="mail-outline"
          iconSize={31.428571428571427}
          iconWidth={62.857142857142854}
          inputPadding={25.142857142857142}
          label="E-mail"
          onBlur={[Function]}
          onChange={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          passiveIconColor="#a3a3a3"
          style={
            Array [
              Object {
                "color": "black",
                "flex": 1,
                "fontSize": 18,
                "padding": 7,
                "paddingLeft": 0,
              },
              Object {
                "color": "#f95a25",
                "marginLeft": 88,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginHorizontal": 31.428571428571427,
          "marginTop": 15.714285714285714,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "overflow": "hidden",
              "paddingTop": 16,
            },
            undefined,
            Object {
              "height": 73.14285714285714,
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#f95a25",
                "fontSize": 31.428571428571427,
              },
              Object {
                "bottom": 24,
                "color": "rgba(163, 163, 163, 1)",
                "left": 25.142857142857142,
                "position": "absolute",
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#f0f0f0",
                "position": "absolute",
                "top": 8,
                "width": 1,
              },
              Object {
                "height": 48,
                "left": 70.85714285714286,
              },
            ]
          }
        />
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 48,
              "left": 88,
              "position": "absolute",
              "top": 24,
            }
          }
        >
          <Text
            collapsable={false}
            style={
              Object {
                "color": "rgba(105, 105, 105, 1)",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Password
          </Text>
        </View>
        <TextInput
          animationDuration={300}
          height={48}
          iconClass={[Function]}
          iconColor="#f95a25"
          iconName="lock-closed-outline"
          iconSize={31.428571428571427}
          iconWidth={62.857142857142854}
          inputPadding={25.142857142857142}
          label="Password"
          onBlur={[Function]}
          onChange={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          passiveIconColor="#a3a3a3"
          secureTextEntry={true}
          style={
            Array [
              Object {
                "color": "black",
                "flex": 1,
                "fontSize": 18,
                "padding": 7,
                "paddingLeft": 0,
              },
              Object {
                "color": "#f95a25",
                "marginLeft": 88,
              },
              undefined,
            ]
          }
          underlineColorAndroid="transparent"
        />
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#659cff",
          "borderRadius": 7.857142857142857,
          "marginHorizontal": 31.428571428571427,
          "marginTop": 15.714285714285714,
          "opacity": 1,
          "paddingVertical": 23.57142857142857,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 25.142857142857142,
            "textAlign": "center",
          }
        }
      >
        Register
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 7.857142857142857,
          "marginTop": 15.714285714285714,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontFamily": "Poppins-Regular",
            "fontSize": 22,
          }
        }
      >
        Already have an account?
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "blue",
            "fontSize": 25.142857142857142,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
